File: src/main/java/github/ptrteixeira/Standing.java
package github.ptrteixeira;

import javafx.beans.property.SimpleStringProperty;

/**
 * JavaBean wrapper for sports win/loss records records.
 * 
 * In context, used to generate relative standings between teams by sorting 
 * by their win/loss records.
 * 
 * @author Peter
 */
public class Standing {
    private final SimpleStringProperty teamName;
    private final SimpleStringProperty conference;
    private final SimpleStringProperty overall;
    
    public Standing(String teamName, String conference, String overall) {
        this.teamName = new SimpleStringProperty(teamName);
        this.conference = new SimpleStringProperty(conference);
        this.overall = new SimpleStringProperty(overall);
    }
    
    public String getTeamName() {
        return teamName.get();
    }
    
    public void setTeamName(String teamName) {
        this.teamName.set(teamName);
    }
    
    public String getConference() {
        return conference.get();
    }
    
    public void setConference(String conference) {
        this.conference.set(conference);
    }
    
    public String getOverall() {
        return overall.get();
    }
    
    public void setOverall(String overall) {
        this.overall.set(overall);
    }
}

File: src/main/kotlin/com/github/ptrteixeira/nusports/model/Standing.kt
/* Released under the MIT license, 2018 */

package com.github.ptrteixeira.nusports.model

/**
 * Data object to record sports win/loss ratios
 *
 * In context, used to generate relative standings between teams based on their win/loss
 * records.
 *
 * @property teamName Name of the team whose standings this records. eg, "Northeastern"
 * @property conference Record within conference play, eg "5-2"
 * @property overall Overall record, including both in-conference and out-of conference play
 */
data class Standing(
    val teamName: String,
    val conference: String,
    val overall: String
)
